service/history/workflow/context.go

```diff
@@ -873,11 +873,15 @@ func (c *ContextImpl) ReapplyEvents(
 	return err
 }
 
-func (c *ContextImpl) UpdateRegistry(ctx context.Context) update.Registry {
+func (c *ContextImpl) UpdateRegistry(ctx context.Context, ms MutableState) update.Registry {
 	if c.updateRegistry == nil {
-		nsIDStr := c.MutableState.GetNamespaceEntry().ID().String()
+		if ms == nil {
+			ms = c.MutableState
+		}
+
+		nsIDStr := ms.GetNamespaceEntry().ID().String()
 		c.updateRegistry = update.NewRegistry(
-			func() update.Store { return c.MutableState },
+			func() update.Store { return ms },
```

Comment:
This is awkward; but we don't want to put the MS into the WorkflowLease or have a `SetMutableState` option on the Context either.

Reply:
Also better convert this in `TODO`.